code/obj/item/misc_junk.dm

```diff
@@ -721,3 +721,119 @@ TYPEINFO(/obj/item/reagent_containers/vape)
 	New()
 		..()
 		BLOCK_SETUP(BLOCK_ROPE)
+
+/obj/item/cash_briefcase
+	name = "Briefcase full of cash"
+	desc = "A foldable briefcase that can hold a large amount of cash. "
+	icon = 'icons/obj/items/storage.dmi'
+	icon_state = "briefcase"
+	inhand_image_icon = 'icons/mob/inhand/hand_general.dmi'
+	item_state = "briefcase"
+	w_class = W_CLASS_BULKY
+
+	var/balance = 0 // Current amount of cash in the case
+	var/maximum_balance = 30000 // Maximum amount of cash in the case
+	var/open = FALSE // Is the briefcase open?
+	var/open_icon_state = "briefcase-open"
+	var/closed_icon_state = "briefcase"
+
+	get_desc()
+		. = ""
+		if(src.balance > 0)
+			.+= "It has [src.balance] credits inside!"
+		else
+			.+= "It's empty!"
+
+	update_icon()
+		. = ..()
+		var/overlay_icon_state = null
+		var/image/cashoverlay = null
+		if (src.open)
+			src.icon_state = open_icon_state
+			switch (src.balance)
+				if (12 to 119)
+					overlay_icon_state = "briefcashgreen"
+				if (120 to 1199)
+					overlay_icon_state = "briefcashblue"
+				if (1200 to 5999)
+					overlay_icon_state = "briefcashindi"
+				if (6000 to 11999)
+					overlay_icon_state = "briefcashpurp"
+				if (12000 to INFINITY)
+					overlay_icon_state = "briefcashred"
+				else
+					overlay_icon_state = null
+			if (overlay_icon_state)
+				cashoverlay = image(src.icon, overlay_icon_state)
+		else
+			src.icon_state = closed_icon_state
+		src.UpdateOverlays(cashoverlay, "cash_overlay")
+
+	attack_self(mob/user)
+		src.toggleCase(user)
+
+	attackby(obj/item/W, mob/user)
+		if(istype(W, /obj/item/currency/spacecash))
+			if (!src.open)
+				boutput(user, "You need to open the briefcase to put cash in it.")
+				return
+			else if (src.balance >= src.maximum_balance)
+				boutput(user, "The briefcase is full!")
+				return
+			else
+				var/space_left = src.maximum_balance - src.balance
+				var/obj/item/currency/spacecash/cashstack = W
+				if (space_left < cashstack.amount)
+					src.balance += space_left
+					var/money_to_destroy = cashstack.split_stack(space_left)
+					qdel(money_to_destroy)
+				else
+					src.balance += cashstack.amount
+					user.u_equip(W)
+					qdel(W)
+				src.UpdateIcon()
+		else
+			. = ..()
+
+	attack_hand(mob/user)
+		if (src.open)
+			var/amount = round(input("How much cash do you want to take from the briefcase?") as null|num)
+			if (isnum_safe(amount))
+				if (amount > src.balance || amount < 1)
+					boutput(user, SPAN_ALERT("You wish!"))
+					return
```

Comment:
This should use `tgui_input_number` and dynamically assign a max input value based on the available stack for a better user experience